src/publish.js

```diff
@@ -28,6 +28,35 @@ async function publishOutput() {
   const OUTPUT_BRANCH_SUFFIX = core.getInput("output-branch-suffix");
   if (GITHUB_EVENT_NAME.includes("pull_request")) {
     const PR_NUMBER = github.context.payload.pull_request.number;
+
+    // Build PDF diff
+    if (core.getInput("build-diff-on-pull-request") == "true") {
+      const LATEXDIFF_URL = core.getInput("latexdiff-url");
+      const LATEXPAND_URL = core.getInput("latexpand-url"); 
+      const BASE_REF = github.context.payload.pull_request.base.ref;
+
+      core.startGroup("Build article diff");
+      shell.exec(`cp ${config["ms_pdf"]} .bkup.pdf`);
+
+      // Download latexdiff and latexpand
+      shell.exec(`wget ${LATEXDIFF_URL} && chmod +x latexdiff`);
+      shell.exec(`wget ${LATEXPAND_URL} && chmod +x latexpand`);
+
+      // Checkout base version of ms.tex
+      shell.exec(`./latexpand src/tex/${config["ms_name"]} -o .flat_new.tex`);
+      shell.exec(`git checkout ${BASE_REF} src/tex`);
+      shell.exec(`./latexpand src/tex/${config["ms_name"]} -o .flat_old.tex`);
+
+      // Compute diff, and build
+      shell.exec(`perl latexdiff src/tex/.flat_old.tex src/tex/.flat_new.tex > tmp.tex`);
+      shell.exec(`mv tmp.tex src/tex/ms.tex`);
+      shell.exec(`showyourwork build`);
+      shell.exec(`cp ${config["ms_pdf"]} diff.pdf`);
+      shell.exec(`cp .bkup ${config["ms_pdf"]}`);
+
```

Comment:
I think you want to do an `output.push('diff.pdf')` here.

Reply:
Will do. Why `output.push` the filename here, whereas it is used to push a file containing the filename later?

Reply:
Thanks! Yes it does.